cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/ConstraintResult.java

```diff
@@ -26,7 +26,10 @@
   public static final ConstraintResult SATISFIED = new ConstraintResult(SatisfiedState.SATISFIED, null);
```

Comment:
Can call `new ConstraintResult(SatisfiedState.NEVER_SATISFIED)`, For `ConstraintResult(SatisfiedState satisfiedState, @Nullable Long millisBeforeNextRetry)`, only allow non-null `millisBeforeNextRetry` if the state is  NOT_SATISFIED.